sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/statefulOperators.scala

```diff
@@ -398,6 +427,25 @@ trait StateStoreWriter
     }.toMap
   }
 
+  private def stateStoreInstanceMetrics: Map[String, SQLMetric] = {
+    val provider = StateStoreProvider.create(conf.stateStoreProviderClass)
+    val maxPartitions = conf.defaultNumShufflePartitions
+
+    (0 until maxPartitions).flatMap { partitionId =>
+      provider.supportedInstanceMetrics.flatMap { metric =>
+        val storeNames = if (isJoinOperator) {
+          SymmetricHashJoinStateManager.allStateStoreNames(LeftSide, RightSide)
```

Comment:
There is a better way to do this. Lets not expose Symmetric hash join implementation details here. This is a higher level.

One option is to add a func to `StateStoreWriter` for implementers to provide their state store name. e.g. `stateStoreNames` func returns `DEFAULT_STORE_NAME` by default, but join can override this to provide its own names. That is just one option.

Reply:
This makes sense, and turns out we already have `stateStoreNames` defined the way we want it in StreamingSymmetricHashJoinExec 👍